// @beefree.io/sdk/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.beeTypes = void 0;
var load_script_1 = __importDefault(require("load-script"));
var function_1 = require("fp-ts/lib/function");
var E = __importStar(require("fp-ts/lib/Either"));
var Constants_1 = __importStar(require("./utils/Constants"));
var api_1 = require("./services/api");
var utils_1 = require("./utils/utils");
var beeTypes = __importStar(require("./types/bee"));
exports.beeTypes = beeTypes;
var beeLoaderUrl = '';
var LOAD = Constants_1.default.LOAD, SAVE = Constants_1.default.SAVE, SEND = Constants_1.default.SEND, PREVIEW = Constants_1.default.PREVIEW, SAVE_AS_TEMPLATE = Constants_1.default.SAVE_AS_TEMPLATE, TOGGLE_STRUCTURE = Constants_1.default.TOGGLE_STRUCTURE, TOGGLE_COMMENTS = Constants_1.default.TOGGLE_COMMENTS, TOGGLE_PREVIEW = Constants_1.default.TOGGLE_PREVIEW, TOGGLE_MERGETAGS_PREVIEW = Constants_1.default.TOGGLE_MERGETAGS_PREVIEW, SHOW_COMMENT = Constants_1.default.SHOW_COMMENT, RELOAD = Constants_1.default.RELOAD, LOAD_WORKSPACE = Constants_1.default.LOAD_WORKSPACE, LOAD_STAGE_MODE = Constants_1.default.LOAD_STAGE_MODE, LOAD_CONFIG = Constants_1.default.LOAD_CONFIG, LOAD_ROWS = Constants_1.default.LOAD_ROWS, UPDATE_TOKEN = Constants_1.default.UPDATE_TOKEN, GET_CONFIG = Constants_1.default.GET_CONFIG, SWITCH_TEMPLATE_LANGUAGE = Constants_1.default.SWITCH_TEMPLATE_LANGUAGE, SWITCH_PREVIEW = Constants_1.default.SWITCH_PREVIEW, EXEC_COMMAND = Constants_1.default.EXEC_COMMAND, GET_TEMPLATE_JSON = Constants_1.default.GET_TEMPLATE_JSON;
var Bee = /** @class */ (function () {
    function Bee(token, urlConfig) {
        var _this = this;
        var _a;
        this.bee = function (bee) {
            (0, load_script_1.default)(beeLoaderUrl, function (err) {
                if (err) {
                    throw new Error('BeePlugin.js is not reachable');
                }
                return bee();
            });
        };
        this.UNSAFE_getToken = function (clientId, clientSecret, uid, urlConfig) {
            var _a, _b;
            var localUrlConfig = {
                authUrl: (_a = urlConfig === null || urlConfig === void 0 ? void 0 : urlConfig.authUrl) !== null && _a !== void 0 ? _a : Constants_1.API_AUTH_URL,
                beePluginUrl: (_b = urlConfig === null || urlConfig === void 0 ? void 0 : urlConfig.beePluginUrl) !== null && _b !== void 0 ? _b : Constants_1.BEEJS_URL,
            };
            if (_this.token && _this.token.access_token) {
                throw new Error('Token already declared');
            }
            return (0, api_1.fetchToken)({ authUrl: localUrlConfig.authUrl, clientId: clientId, clientSecret: clientSecret, uid: uid })
                .then(function (res) {
                _this.token = res.data;
                return res.data;
            });
        };
        this.start = function (config, template, //user can pass an empty object in some specific cases
        bucketDir, options) {
            var _a = _this, bee = _a.bee, token = _a.token;
            return (0, function_1.pipe)((0, utils_1.eitherCheckStartParams)(config, token), E.fold(function (_a) {
                var message = _a.message;
                return new Promise(function (reject) { return reject(message); });
            }, function () { return new Promise(function (resolve) {
                bee(function () { return BeePlugin.create(token, __assign(__assign({}, config), { startOrigin: "[npm] ".concat(process.env.NPM_PACKAGE_NAME), startOriginVersion: process.env.NPM_PACKAGE_VERSION }), function (instance) {
                    _this.instance = instance;
                    instance.start(template, options);
                    resolve(instance);
                }, bucketDir); });
            }); }));
        };
        this.startFileManager = function (config, bucketDir, options) {
            var _a = _this, bee = _a.bee, token = _a.token;
            return (0, function_1.pipe)((0, utils_1.eitherCheckStartParams)(config, token), E.fold(function (_a) {
                var message = _a.message;
                return new Promise(function (reject) { return reject(message); });
            }, function () { return new Promise(function (resolve) {
                bee(function () { return BeePlugin.create(token, __assign(__assign({}, config), { startOrigin: "[npm] ".concat(process.env.NPM_PACKAGE_NAME), startOriginVersion: process.env.NPM_PACKAGE_VERSION }), function (instance) {
                    _this.instance = instance;
                    instance.start(options);
                    resolve(instance);
                }, bucketDir); });
            }); }));
        };
        this.join = function (config, sessionId, bucketDir) {
            var _a = _this, bee = _a.bee, token = _a.token;
            return (0, function_1.pipe)((0, utils_1.eitherCheckJoinParams)(config, sessionId, _this.token), E.fold(function (_a) {
                var message = _a.message;
                throw new Error(message);
            }, function () { return new Promise(function (resolve) {
                bee(function () { return BeePlugin.create(token, config, function (instance) {
                    _this.instance = instance;
                    instance.join(sessionId);
                    resolve(instance);
                }, bucketDir); });
            }); }));
        };
        this.executeAction = function (action, param, options) {
            if (param === void 0) { param = {}; }
            if (options === void 0) { options = {}; }
            var instance = _this.instance;
            return (0, function_1.pipe)((0, utils_1.eitherCanExecuteAction)(instance, action), E.fold(function (_a) {
                var message = _a.message;
                throw new Error(message);
            }, function () { return instance[action](param, options); }));
        };
        this.executeGetConfigAction = function () {
            var instance = _this.instance;
            return instance[GET_CONFIG]();
        };
        this.load = function (template) { return _this.executeAction(LOAD, template); };
        this.loadRows = function () { return _this.executeAction(LOAD_ROWS); };
        this.save = function (options) { return _this.executeAction(SAVE, options); };
        this.saveAsTemplate = function () { return _this.executeAction(SAVE_AS_TEMPLATE); };
        this.send = function (args) { return _this.executeAction(SEND, args); };
        this.preview = function () { return _this.executeAction(PREVIEW); };
        this.toggleStructure = function () { return _this.executeAction(TOGGLE_STRUCTURE); };
        this.togglePreview = function () { return _this.executeAction(TOGGLE_PREVIEW); };
        this.toggleComments = function () { return _this.executeAction(TOGGLE_COMMENTS); };
        this.toggleMergeTagsPreview = function () { return _this.executeAction(TOGGLE_MERGETAGS_PREVIEW); };
        this.showComment = function (comment) { return _this.executeAction(SHOW_COMMENT, comment); };
        this.reload = function (template, options) { return _this.executeAction(RELOAD, template, options); };
        this.loadWorkspace = function (type) { return _this.executeAction(LOAD_WORKSPACE, type); };
        this.loadStageMode = function (args) { return _this.executeAction(LOAD_STAGE_MODE, args); };
        this.loadConfig = function (args) { return _this.executeAction(LOAD_CONFIG, args); };
        this.updateToken = function (updateTokenArgs) { return _this.executeAction(UPDATE_TOKEN, updateTokenArgs); };
        this.getConfig = function () { return _this.executeGetConfigAction(); };
        this.switchTemplateLanguage = function (args) { return _this.executeAction(SWITCH_TEMPLATE_LANGUAGE, args); };
        this.switchPreview = function (args) { return _this.executeAction(SWITCH_PREVIEW, args); };
        this.execCommand = function (command, options) { return _this.executeAction(EXEC_COMMAND, command, options); };
        this.getTemplateJson = function () {
            return _this.executeAction(GET_TEMPLATE_JSON, {});
        };
        beeLoaderUrl = (_a = urlConfig === null || urlConfig === void 0 ? void 0 : urlConfig.beePluginUrl) !== null && _a !== void 0 ? _a : Constants_1.BEEJS_URL;
        this.token = token || Constants_1.mockedEmptyToken;
        this.instance = null;
    }
    return Bee;
}());
exports.default = Bee;
