// @beefree.io/sdk/dist/types/bee.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PREVIEW_CONTROL = exports.TokenStatus = exports.BeePluginRoles = exports.EventCodes = exports.ActionCodes = exports.ContentCodes = exports.WorkspaceStage = exports.OnCommentChangeEnum = exports.EngageHandle = exports.RowLayoutType = exports.ModuleDescriptorOrderNames = exports.ModuleDescriptorNames = exports.ModuleTypes = exports.OnInfoDetailHandle = exports.BeePluginErrorCodes = exports.LoadWorkspaceOptions = exports.ExecCommands = exports.StageDisplayOptions = exports.StageModeOptions = void 0;
var StageModeOptions;
(function (StageModeOptions) {
    StageModeOptions["DESKTOP"] = "desktop";
    StageModeOptions["MOBILE"] = "mobile";
})(StageModeOptions = exports.StageModeOptions || (exports.StageModeOptions = {}));
var StageDisplayOptions;
(function (StageDisplayOptions) {
    StageDisplayOptions["BLUR"] = "blur";
    StageDisplayOptions["HIDE"] = "hide";
})(StageDisplayOptions = exports.StageDisplayOptions || (exports.StageDisplayOptions = {}));
var ExecCommands;
(function (ExecCommands) {
    ExecCommands["SELECT"] = "select";
    ExecCommands["SCROLL"] = "scroll";
    ExecCommands["HIGHLIGHT"] = "highlight";
    ExecCommands["FOCUS"] = "focus";
})(ExecCommands = exports.ExecCommands || (exports.ExecCommands = {}));
var LoadWorkspaceOptions;
(function (LoadWorkspaceOptions) {
    LoadWorkspaceOptions["DEFAULT"] = "default";
    LoadWorkspaceOptions["MIXED"] = "mixed";
    LoadWorkspaceOptions["AMP_ONLY"] = "amp_only";
    LoadWorkspaceOptions["HTML_ONLY"] = "html_only";
})(LoadWorkspaceOptions = exports.LoadWorkspaceOptions || (exports.LoadWorkspaceOptions = {}));
var BeePluginErrorCodes;
(function (BeePluginErrorCodes) {
    BeePluginErrorCodes[BeePluginErrorCodes["SAVE_UNAVAILABLE"] = 1000] = "SAVE_UNAVAILABLE";
    BeePluginErrorCodes[BeePluginErrorCodes["AMP_CONTENT_DETECTED"] = 1001] = "AMP_CONTENT_DETECTED";
    BeePluginErrorCodes[BeePluginErrorCodes["INVALID_TEMPLATES"] = 1100] = "INVALID_TEMPLATES";
    BeePluginErrorCodes[BeePluginErrorCodes["TEMPLATE_CANNOT_SAVED"] = 1200] = "TEMPLATE_CANNOT_SAVED";
    BeePluginErrorCodes[BeePluginErrorCodes["LOCKED_ROW_CLICKED"] = 1300] = "LOCKED_ROW_CLICKED";
    BeePluginErrorCodes[BeePluginErrorCodes["LOCKED_MODULE_CLICKED"] = 1310] = "LOCKED_MODULE_CLICKED";
    BeePluginErrorCodes[BeePluginErrorCodes["WORKSPACE_NOT_AVAILABLE"] = 1002] = "WORKSPACE_NOT_AVAILABLE";
    BeePluginErrorCodes[BeePluginErrorCodes["FEATURE_NOT_AVAILABLE_FOR_PLAN"] = 1704] = "FEATURE_NOT_AVAILABLE_FOR_PLAN";
    BeePluginErrorCodes[BeePluginErrorCodes["GENERIC_BUMP_ERROR"] = 2000] = "GENERIC_BUMP_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["INVALID_TARGET_VERSION"] = 2100] = "INVALID_TARGET_VERSION";
    BeePluginErrorCodes[BeePluginErrorCodes["VALIDATION_ERROR_DETAIL"] = 2200] = "VALIDATION_ERROR_DETAIL";
    BeePluginErrorCodes[BeePluginErrorCodes["MISSING_TEMPLATE_VERSION"] = 2300] = "MISSING_TEMPLATE_VERSION";
    BeePluginErrorCodes[BeePluginErrorCodes["INVALID_TEMPLATE_VERSION"] = 2400] = "INVALID_TEMPLATE_VERSION";
    BeePluginErrorCodes[BeePluginErrorCodes["TRASFORMATION_ERROR"] = 2500] = "TRASFORMATION_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["BACKWARD_TRASFORMATION_ERROR"] = 2600] = "BACKWARD_TRASFORMATION_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["SERVICE_ERROR"] = 3000] = "SERVICE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["API_GENERIC_ERROR"] = 9900] = "API_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["GENERIC_ERROR"] = 9999] = "GENERIC_ERROR";
    //File System provider errors
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_GENERIC_ERROR"] = 3001] = "FSP_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_BE_GENERIC_ERROR"] = 3100] = "FSP_BE_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_RESOURCE_NOT_FOUND"] = 3200] = "FSP_RESOURCE_NOT_FOUND";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_PERMISSION_DENIED"] = 3300] = "FSP_PERMISSION_DENIED";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_RESOURCE_ALREADY_EXIST"] = 3400] = "FSP_RESOURCE_ALREADY_EXIST";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_FILE_NOT_UPLOADED"] = 3450] = "FSP_FILE_NOT_UPLOADED";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_REQUEST_ERROR"] = 3500] = "FSP_REQUEST_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_USER_ERROR"] = 3600] = "FSP_USER_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["FSP_WRONG_CREDENTIALS"] = 3650] = "FSP_WRONG_CREDENTIALS";
    //Auth Errors
    BeePluginErrorCodes[BeePluginErrorCodes["AUTH_HEADER_MISSING"] = 4001] = "AUTH_HEADER_MISSING";
    BeePluginErrorCodes[BeePluginErrorCodes["AUTH_BEARER_INVALID"] = 4005] = "AUTH_BEARER_INVALID";
    BeePluginErrorCodes[BeePluginErrorCodes["AUTH_TOKEN_EXPIRED"] = 4010] = "AUTH_TOKEN_EXPIRED";
    //Bump error
    BeePluginErrorCodes[BeePluginErrorCodes["BUMP_SERVICE_ERROR"] = 4015] = "BUMP_SERVICE_ERROR";
    //JSON pre-processing errors
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_GENERIC_ERROR"] = 4101] = "JSON_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_CODE_TAG_ERROR"] = 4105] = "JSON_CODE_TAG_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_HTML_BLOCK_ERROR"] = 4110] = "JSON_HTML_BLOCK_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_CONDITIONAL_STATEMENT_ERROR"] = 4115] = "JSON_CONDITIONAL_STATEMENT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_ADD_ROW_CONDITIONAL_STATEMENT_ERROR"] = 4120] = "JSON_ADD_ROW_CONDITIONAL_STATEMENT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_ADD_ROW_STYLE_ERROR"] = 4125] = "JSON_ADD_ROW_STYLE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_ADD_ROW_CLASS_ERROR"] = 4130] = "JSON_ADD_ROW_CLASS_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_ADD_ROW_COMPUTED_STYLE_ERROR"] = 4135] = "JSON_ADD_ROW_COMPUTED_STYLE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_CELL_WIDTH_ERROR"] = 4140] = "JSON_CELL_WIDTH_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_ADD_COLUMN_BACKGROUND_ERROR"] = 4145] = "JSON_ADD_COLUMN_BACKGROUND_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_MAIN_CSS_ERROR"] = 4190] = "JSON_MAIN_CSS_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_MAIN_CSS_MEDIA_ERROR"] = 4191] = "JSON_MAIN_CSS_MEDIA_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_SPECIFIC_CSS_ERROR"] = 4192] = "JSON_SPECIFIC_CSS_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_CSS_RULE_ERROR"] = 4501] = "JSON_CSS_RULE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_CSS_GENERALE_PAGE_ERROR"] = 4502] = "JSON_CSS_GENERALE_PAGE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_BODY_CSS_ERROR"] = 4503] = "JSON_BODY_CSS_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["JSON_COVERTING_CSS_ERROR"] = 4504] = "JSON_COVERTING_CSS_ERROR";
    //HTML creation errors
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_TO_JSON_CONVERT_ERROR"] = 4201] = "HTML_TO_JSON_CONVERT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_TO_JSON__ERROR"] = 4202] = "HTML_COMMUNICATION_TO_JSON__ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_TEXT_ERROR"] = 4210] = "HTML_COMMUNICATION_WORKER_TEXT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_TEXT_ERROR"] = 4211] = "HTML_TEXT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_BUTTON_ERROR"] = 4220] = "HTML_COMMUNICATION_WORKER_BUTTON_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_BUTTON_ERROR"] = 4221] = "HTML_BUTTON_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_VIDEO_ERROR"] = 4230] = "HTML_COMMUNICATION_WORKER_VIDEO_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_VIDEO_ERROR"] = 4231] = "HTML_VIDEO_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_SOCIAL_ERROR"] = 4240] = "HTML_COMMUNICATION_WORKER_SOCIAL_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_SOCIAL_ERROR"] = 4241] = "HTML_SOCIAL_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_ICON_ERROR"] = 4242] = "HTML_COMMUNICATION_WORKER_ICON_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_ICON_ERROR"] = 4243] = "HTML_ICON_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_MENU_ERROR"] = 4244] = "HTML_COMMUNICATION_WORKER_MENU_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_MENU_ERROR"] = 4245] = "HTML_MENU_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_IMAGE_ERROR"] = 4250] = "HTML_COMMUNICATION_WORKER_IMAGE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_IMAGE_ERROR"] = 4251] = "HTML_IMAGE_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_COMMUNICATION_WORKER_MG_ERROR"] = 4260] = "HTML_COMMUNICATION_WORKER_MG_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_CUSTOM_ADDON_ERROR"] = 4270] = "HTML_CUSTOM_ADDON_ERROR";
    //POST-PROCESSING
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_POST_GENERIC_ERROR"] = 4301] = "HTML_POST_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_POST_IMAGE_URI_ERROR"] = 4310] = "HTML_POST_IMAGE_URI_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_POST_CUSTOM_HTML_ERROR"] = 4320] = "HTML_POST_CUSTOM_HTML_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_POST_CUSTOM_CODE_TAG_ERROR"] = 4330] = "HTML_POST_CUSTOM_CODE_TAG_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["HTML_POST_DISPLAY_CONDITION"] = 4340] = "HTML_POST_DISPLAY_CONDITION";
    //RENDERING ERRORS
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_GENERIC_ERROR"] = 4600] = "RENDER_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_CLEAN_HTML_ERROR"] = 4601] = "RENDER_CLEAN_HTML_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_CLEAN_TEXT_ERROR"] = 4602] = "RENDER_CLEAN_TEXT_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_CHECK_FORM_ERROR"] = 4603] = "RENDER_CHECK_FORM_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_VALIDATION_ERROR"] = 4604] = "RENDER_VALIDATION_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["RENDER_VIDEO_FORMAT_NOT_SUPPORTED_ERROR"] = 4605] = "RENDER_VIDEO_FORMAT_NOT_SUPPORTED_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["POST_PROCESSING_FINAL_PAGE_ERROR"] = 4700] = "POST_PROCESSING_FINAL_PAGE_ERROR";
    //SESSION TOKEN ERRORS
    BeePluginErrorCodes[BeePluginErrorCodes["EXPIRED_TOKEN_CANNOT_REFRESHED"] = 5101] = "EXPIRED_TOKEN_CANNOT_REFRESHED";
    BeePluginErrorCodes[BeePluginErrorCodes["EXPIRED_TOKEN_MUST_REFRESHED"] = 5102] = "EXPIRED_TOKEN_MUST_REFRESHED";
    // WEBHOOK ERRORS
    BeePluginErrorCodes[BeePluginErrorCodes["WEBHOOK_CALL_FAILED"] = 5250] = "WEBHOOK_CALL_FAILED";
    // Frontend API errors
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_GENERIC_ERROR"] = 7000] = "FE_API_GENERIC_ERROR";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_INVALID_COMMAND"] = 7005] = "FE_API_INVALID_COMMAND";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_INVALID_COMMAND_OPTIONS"] = 7010] = "FE_API_INVALID_COMMAND_OPTIONS";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_ENTITY_LOCKED"] = 7020] = "FE_API_ENTITY_LOCKED";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_ENTITY_NOT_FOUND"] = 7030] = "FE_API_ENTITY_NOT_FOUND";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_ELEMENT_NOT_FOUND"] = 7040] = "FE_API_ELEMENT_NOT_FOUND";
    BeePluginErrorCodes[BeePluginErrorCodes["FE_API_FORBIDDEN_COMMAND"] = 7050] = "FE_API_FORBIDDEN_COMMAND";
})(BeePluginErrorCodes = exports.BeePluginErrorCodes || (exports.BeePluginErrorCodes = {}));
var OnInfoDetailHandle;
(function (OnInfoDetailHandle) {
    OnInfoDetailHandle["AI_INTEGRATION"] = "ai-integration";
    OnInfoDetailHandle["AI_ALT_TEXT"] = "ai-alt-text";
    OnInfoDetailHandle["AI_IMAGE_GENERATION"] = "ai-image-generation";
})(OnInfoDetailHandle = exports.OnInfoDetailHandle || (exports.OnInfoDetailHandle = {}));
exports.ModuleTypes = {
    DIVIDER: 'mailup-bee-newsletter-modules-divider',
    TEXT: 'mailup-bee-newsletter-modules-text',
    IMAGE: 'mailup-bee-newsletter-modules-image',
    BUTTON: 'mailup-bee-newsletter-modules-button',
    HTML: 'mailup-bee-newsletter-modules-html',
    SOCIAL: 'mailup-bee-newsletter-modules-social',
    EMPTY: 'mailup-bee-newsletter-modules-empty',
    VIDEO: 'mailup-bee-newsletter-modules-video',
    ADDON: 'mailup-bee-newsletter-modules-addon',
    FORM: 'mailup-bee-newsletter-modules-form',
    MERGE_CONTENT: 'mailup-bee-newsletter-modules-merge-content',
    CAROUSEL: 'mailup-bee-newsletter-modules-carousel',
    MENU: 'mailup-bee-newsletter-modules-menu',
    ICONS: 'mailup-bee-newsletter-modules-icons',
    HEADING: 'mailup-bee-newsletter-modules-heading',
    SPACER: 'mailup-bee-newsletter-modules-spacer',
    PARAGRAPH: 'mailup-bee-newsletter-modules-paragraph',
    LIST: 'mailup-bee-newsletter-modules-list',
    TABLE: 'mailup-bee-newsletter-modules-table',
};
exports.ModuleDescriptorNames = {
    DIVIDER: 'divider',
    TEXT: 'text',
    IMAGE: 'image',
    BUTTON: 'button',
    HTML: 'html',
    SOCIAL: 'social',
    VIDEO: 'video',
    ADDON: 'addon',
    FORM: 'form',
    MERGE_CONTENT: 'mergeContent',
    CAROUSEL: 'carousel',
    MENU: 'menu',
    ICONS: 'icons',
    HEADING: 'heading',
    SPACER: 'spacer',
    PARAGRAPH: 'paragraph',
    LIST: 'list'
};
exports.ModuleDescriptorOrderNames = {
    DIVIDER: 'Divider',
    TEXT: 'Text',
    IMAGE: 'Image',
    BUTTON: 'Button',
    HTML: 'Html',
    SOCIAL: 'Social',
    VIDEO: 'Video',
    FORM: 'Form',
    MERGE_CONTENT: 'MergeContent',
    CAROUSEL: 'Carousel',
    MENU: 'Menu',
    ICONS: 'Icons',
    HEADING: 'Heading',
    SPACER: 'Spacer',
    PARAGRAPH: 'Paragraph',
    LIST: 'List'
};
exports.RowLayoutType = {
    ONE_COLUMNS_EMPTY: 'one-column-empty',
    TWO_COLUMNS_EMPTY: 'two-columns-empty',
    TWO_COLUMNS_4_8_EMPTY: 'two-columns-4-8-empty',
    TWO_COLUMNS_8_4_EMPTY: 'two-columns-8-4-empty',
    TWO_COLUMNS_3_9_EMPTY: 'two-columns-3-9-empty',
    TWO_COLUMNS_9_3_EMPTY: 'two-columns-9-3-empty',
    THREE_COLUMNS_EMPTY: 'three-columns-empty',
    THREE_COLUMNS_3_3_6_EMPTY: 'three-columns-3-3-6-empty',
    THREE_COLUMNS_3_6_3_EMPTY: 'three-columns-3-6-3-empty',
    THREE_COLUMNS_6_3_3_EMPTY: 'three-columns-6-3-3-empty',
    FOUR_COLUMNS_EMPTY: 'four-columns-empty',
    SIX_COLUMNS_EMPTY: 'six-columns-empty',
};
var EngageHandle;
(function (EngageHandle) {
    EngageHandle["MDM"] = "mdm";
})(EngageHandle = exports.EngageHandle || (exports.EngageHandle = {}));
var OnCommentChangeEnum;
(function (OnCommentChangeEnum) {
    OnCommentChangeEnum["NEWCOMMENT"] = "NEW_COMMENT";
    OnCommentChangeEnum["DELETECOMMENT"] = "COMMENT_DELETED";
    OnCommentChangeEnum["CHANGECOMMENT"] = "COMMENT_EDITED";
    OnCommentChangeEnum["THREADRESOLVED"] = "COMMENT_THREAD_RESOLVED";
    OnCommentChangeEnum["THREADREOPENED"] = "COMMENT_THREAD_REOPENED";
})(OnCommentChangeEnum = exports.OnCommentChangeEnum || (exports.OnCommentChangeEnum = {}));
var WorkspaceStage;
(function (WorkspaceStage) {
    WorkspaceStage["desktop"] = "desktop";
    WorkspaceStage["mobile"] = "mobile";
    WorkspaceStage["global"] = "global";
})(WorkspaceStage = exports.WorkspaceStage || (exports.WorkspaceStage = {}));
exports.ContentCodes = {
    TEXT_BLOCK: '01',
    IMAGE_BLOCK: '02',
    BUTTON_BLOCK: '03',
    DIVIDER_BLOCK: '04',
    SOCIAL_BLOCK: '05',
    DYNAMIC_CONTENT_BLOCK: '06',
    HTML_BLOCK: '07',
    VIDEO_BLOCK: '08',
    FORM: '09',
    ICONS: '10',
    MENU: '11',
    ROW: '14',
    MESSAGE: '16',
    SPACER: '18',
    PARAGRAPH: '22',
    LIST: '23',
    TABLE: '26',
};
exports.ActionCodes = {
    DROPPED: '00',
    DRAGGED: '01',
    DELETED: '02',
    DUPLICATED: '03',
    CHANGED: '04',
    OPENED: '05',
    CLOSED: '06',
    LOCKED: '07',
    SAVED: '08',
    RESTORED: '09',
    CONTENT_AREA_BACKGROUND_COLOR: '10',
    DO_NOT_STACK_ON_MOBILE: '11',
    ROW_BACKGROUND_IMAGE: '12',
    BACKGROUND_CENTER: '13',
    BACKGROUND_REPEAT: '14',
    BACKGROUND_FULL_WIDTH: '15',
    ROW_DISPLAY_CONDITION: '16',
    REVERSE_STACK_ORDER_ON_MOBILE: '17',
    TEXT_COLOR: '20',
    LINK_COLOR: '21',
    TEXT_EDITED: '23',
    LINE_HEIGHT: '24',
    CONTENT_AREA_WIDTH: '25',
    BACKGROUND_COLOR: '27',
    DEFAULT_FONT: '28',
    PADDING_ALL_SIDES: '30',
    PADDING_LEFT: '31',
    PADDING_RIGHT: '32',
    PADDING_TOP: '33',
    PADDING_BOTTOM: '34',
    HIDE_ON_MOBILE: '40',
    VIDEO_URL: '41',
    PLAY_ICON_TYPE: '42',
    PLAY_ICON_COLOR: '43',
    PLAY_ICON_SIZE: '44',
    ALIGN: '50',
    AUTOMATIC_IMAGE_RESIZING: '51',
    FULL_WIDTH_ON_MOBILE: '52',
    IMAGE_WIDTH: '53',
    ALTERNATE_TEXT: '60',
    DYNAMIC_IMAGE_SRC: '61',
    DYNAMIC_IMAGE_TOGGLE: '62',
    CHANGE_IMAGE: '63',
    IMAGE_LINK: '64',
    BUTTON_ALIGN: '70',
    BUTTON_LINK_TYPE: '71',
    BUTTON_WIDTH: '72',
    BUTTON_AUTO_WIDTH: '73',
    BUTTON_BACKGROUND_COLOR: '74',
    BORDER_RADIUS: '75',
    HTML_EDITED: '80',
    BORDER_ALL_SIDES: '81',
    BORDER_LEFT: '82',
    BORDER_RIGHT: '83',
    BORDER_TOP: '84',
    BORDER_BOTTOM: '85',
    DIVIDER_LINE_TOGGLE: '90',
    DIVIDER_WIDTH: '91',
    DIVIDER_HEIGHT: '92',
    DIVIDER_ALIGN: '93',
    ICON_NAME: '95',
    ICON_ALTERNATE_TEXT: '96',
    ICON_URL: '97',
    ICON_SPACING: '98',
    ICON_ALIGN: '99',
    BACKGROUND_VIDEO: '128',
    PARAGRAPH_SPACING: '129',
    FONT_WEIGHT: '130',
    LIST_TYPE: '131',
    START_LIST: '132',
    LIST_SPACING: '133',
    LIST_INDENT: '134',
    LIST_STYLE_POSITION: '135',
};
exports.EventCodes = Object.fromEntries(Object.entries(exports.ContentCodes).flatMap(function (_a) {
    var contentKey = _a[0], contentCode = _a[1];
    return Object.entries(exports.ActionCodes).map(function (_a) {
        var actionKey = _a[0], actionCode = _a[1];
        return [
            "".concat(contentKey, "_").concat(actionKey),
            "".concat(contentCode).concat(actionCode),
        ];
    });
}));
var BeePluginRoles;
(function (BeePluginRoles) {
    BeePluginRoles["REVIEWER"] = "reviewer";
})(BeePluginRoles = exports.BeePluginRoles || (exports.BeePluginRoles = {}));
var TokenStatus;
(function (TokenStatus) {
    TokenStatus["OK"] = "ok";
    TokenStatus["REFRESHING"] = "refreshing";
})(TokenStatus = exports.TokenStatus || (exports.TokenStatus = {}));
exports.PREVIEW_CONTROL = {
    DARK_MODE: 'dark',
    AMP: 'amp',
    LANGUAGE: 'language',
    DEVICE: 'device',
    SIZE: 'size',
    DISPLAY_CONDITIONS: 'DISPLAY_CONDITIONS',
};
